### PR TITLE
Jsdoc property description

### PR DESCRIPTION
**Pull request by danay1999**
Fixes [#47933](https://github.com/microsoft/TypeScript/issues/47933)

### CODE REVIEW

File: src/services/symbolDisplay.ts [hunk 1]
```diff
@@ -638,26 +648,50 @@ namespace ts.SymbolDisplay {
...
+                                fullSymbolDisplayParts[index++] = spacePart();
+                                fullSymbolDisplayParts[index++] = punctuationPart(SyntaxKind.BarToken);
+                                fullSymbolDisplayParts[index++] = spacePart();
+                            }
+                        }
+                        fullSymbolDisplayParts[index] = punctuationPart(SyntaxKind.CloseBracketToken);
+                    }
+                }
+                else{
+                    fullSymbolDisplayParts[2] = textOrKeywordPart(TypeFlags[`1`]); //This is the fallback in case else fails
```
Comment (weswigham) [inline src/services/symbolDisplay.ts:679]: ...? `TypeFlags[1]` doesn't sound like any keyword I know. What's it supposed to be referring to?

Comment (danay1999) [reply src/services/symbolDisplay.ts:679]: returns `Any`

Comment (weswigham) [reply src/services/symbolDisplay.ts:677]: Should probably manually write out `"Any"` then.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -42214,6 +42215,19 @@ namespace ts {
...
+                            const expressionType = checkExpressionCached(name.expression);
+                            let infos  = getApplicableIndexInfos(expressionType, getLiteralTypeFromPropertyName(name.name));
+                            if(infos.length === 0 && getIndexInfosOfType(expressionType).filter(type => type.keyType.flags & TypeFlags.TemplateLiteral || type.keyType.flags & TypeFlags.ESSymbol).length > 0){
+                                infos = getApplicableIndexInfos(expressionType, checkExpressionCached(name.name));
+                            }
+                            if (length(infos) && infos[0].declaration && infos[0].declaration?.symbol.flags & SymbolFlags.Signature && infos[0].declaration?.jsDoc) {
+                                const copy = createSymbol(SymbolFlags.Signature, InternalSymbolName.Index);
+                                copy.declarations = mapDefined(infos, i => i.declaration);
+                                copy.parent = expressionType.symbol ? expressionType.symbol : getSymbolAtLocation(copy.declarations[0].parent);
+                                return copy;
```
Comment (RyanCavanaugh) [inline src/compiler/checker.ts:42228]: Need to assign to `link.resolvedSymbol` before returning

**Pull request closed without merging**